internal/datanode/services.go

```diff
@@ -578,3 +583,239 @@ func (node *DataNode) DropCompactionPlan(ctx context.Context, req *datapb.DropCo
 	log.Ctx(ctx).Info("DropCompactionPlans success", zap.Int64("planID", req.GetPlanID()))
 	return merr.Success(), nil
 }
+
+// CreateTask creates different types of tasks based on task type
+func (node *DataNode) CreateTask(ctx context.Context, request *workerpb.CreateTaskRequest) (*commonpb.Status, error) {
+	log.Ctx(ctx).Info("CreateTask received", zap.Any("properties", request.GetProperties()))
+	if err := merr.CheckHealthy(node.GetStateCode()); err != nil {
+		return merr.Status(err), nil
+	}
+	properties := taskcommon.NewProperties(request.GetProperties())
+	taskType, err := properties.GetTaskType()
+	if err != nil {
+		return merr.Status(err), nil
+	}
+	switch taskType {
+	case taskcommon.PreImport:
+		req := &datapb.PreImportRequest{}
+		err := proto.Unmarshal(request.GetPayload(), req)
+		if err != nil {
+			return merr.Status(err), nil
+		}
+		return node.PreImport(ctx, req)
+	case taskcommon.Import:
+		req := &datapb.ImportRequest{}
+		err := proto.Unmarshal(request.GetPayload(), req)
+		if err != nil {
+			return merr.Status(err), nil
+		}
+		return node.ImportV2(ctx, req)
+	case taskcommon.Compaction:
+		req := &datapb.CompactionPlan{}
+		err := proto.Unmarshal(request.GetPayload(), req)
+		if err != nil {
+			return merr.Status(err), nil
+		}
+		return node.CompactionV2(ctx, req)
+	case taskcommon.Index:
+		req := &workerpb.CreateJobRequest{}
+		err := proto.Unmarshal(request.GetPayload(), req)
+		if err != nil {
+			return merr.Status(err), nil
+		}
+		return node.CreateJobV2(ctx, &workerpb.CreateJobV2Request{
```

Comment:
Discontinue the use of the outdated CreateJobV2 and instead decompose it into specific xxxTasks.

Reply:
Ok, done